File: 2026POTD/JANUARY/11-01-2026/85.swift
class Solution {
    func maximalRectangle(_ matrix: [[Character]]) -> Int {
        if matrix.isEmpty { return 0 }

        let cols = matrix[0].count
        var heights = Array(repeating: 0, count: cols)
        var ans = 0

        for row in matrix {
            for j in 0..<cols {
                heights[j] = row[j] == "1" ? heights[j] + 1 : 0
            }

            var stack: [Int] = []
            for i in 0...cols {
                let h = (i == cols) ? 0 : heights[i]
                while !stack.isEmpty && h < heights[stack.last!] {
                    let height = heights[stack.removeLast()]
                    let width = stack.isEmpty ? i : i - stack.last! - 1
                    ans = max(ans, height * width)
                }
                stack.append(i)
            }
        }

        return ans
    }
}

File: 2026POTD/JANUARY/11-01-2026/85.dart
class Solution {
  int maximalRectangle(List<List<String>> matrix) {
    if (matrix.isEmpty) return 0;

    int cols = matrix[0].length;
    List<int> heights = List.filled(cols, 0);
    int ans = 0;

    for (var row in matrix) {
      for (int j = 0; j < cols; j++) {
        heights[j] = row[j] == '1' ? heights[j] + 1 : 0;
      }

      List<int> stack = [];
      for (int i = 0; i <= cols; i++) {
        int h = i == cols ? 0 : heights[i];
        while (stack.isNotEmpty && h < heights[stack.last]) {
          int height = heights[stack.removeLast()];
          int width = stack.isEmpty ? i : i - stack.last - 1;
          ans = ans > height * width ? ans : height * width;
        }
        stack.add(i);
      }
    }

    return ans;
  }
}